platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/pax/web/jetty/JettyAuthenticator.java

```diff
@@ -112,15 +106,47 @@ public void doFilter(
         if (!hasBeenInitialized(securityFilterServiceReference, bundleContext)) {
           initializeSecurityFilter(bundleContext, securityFilterServiceReference, securityFilter);
         }
-
         chain.addSecurityFilter(securityFilter);
       }
 
-      chain.doFilter(servletRequest, servletResponse);
+      try {
+        chain.doFilter(servletRequest, servletResponse);
+      } catch (IOException e) {
+        throw new ServerAuthException(
+            "Unable to process security filter. Blocking the request processing.");
+      } catch (AuthenticationChallengeException e) {
+        return new Authentication.Challenge() {};
+      } catch (AuthenticationException e) {
+        return new Authentication.Failure() {};
+      }
     } else {
-      LOGGER.debug("Did not find any SecurityFilters. Acting as a pass-through filter...");
-      filterChain.doFilter(servletRequest, servletResponse);
+      LOGGER.debug("Did not find any SecurityFilters. Send auth failure...");
+      return new Authentication.Failure() {};
+    }
+
+    Subject subject = (Subject) servletRequest.getAttribute(SecurityConstants.SECURITY_SUBJECT);
+    UserIdentity userIdentity = new JettyUserIdentity(getSecuritySubject(subject));
+    return new JettyAuthenticatedUser(userIdentity);
+  }
+
+  @Nullable
+  protected BundleContext getContext() {
+    final Bundle cxfBundle = FrameworkUtil.getBundle(JettyAuthenticator.class);
+    if (cxfBundle != null) {
+      return cxfBundle.getBundleContext();
+    }
+    return null;
+  }
+
+  @Nullable
+  private javax.security.auth.Subject getSecuritySubject(@Nullable Subject subject) {
+    if (subject == null) {
+      return null;
     }
+    HashSet emptySet = new HashSet();
+    HashSet subjectPrincipal = new HashSet();
+    subjectPrincipal.add(subject);
+    return new javax.security.auth.Subject(true, subjectPrincipal, emptySet, emptySet);
```

Comment:
✏️ It looks like we might be able to use `Collections.emptyMap()` here instead of `emptySet`